keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,635 @@
+severity of the issue.
+If the issue is low severity evaluate if the the backport can be skipped.
+If the issue is high servery consider a backport.
+
+### Risks and Mitigations
+
+#### A portion of the kubeadm backend changes and breaks users
+
+As mentioned in "User story 3", the kubeadm maintainers
+will give no guarantees about the kubeadm backend until it is
+announced as stable. README files in the kubeadm repository
+should denote that.
+
+It is out of scope for this proposal to try to solve the problem.
+
+#### The post-submit job for tagging/branching k/kubeadm fails
+
+In cases the post-submit job fails, maintainers should be able
+to apply the required changes manually or ask Prow maintainers
+to re-trigger the job. Sufficient number of kubeadm maintainers
+should have access and know how to perform these actions.
+
+Ideally the post-submit job should be setup to send an email
+notification on every failure at the SIG Cluster Lifecycle
+mailing list with a custom `testgrid` suffix:
+`kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com`
+This suffix is already used by kubeadm periodic jobs.
+
+Close to a release, coordination with SIG Release will be required,
+to make sure that the exact, desired SHA of kubeadm is released.
+
+#### The kubeadm go.mod includes outdated components
+
+Once kubeadm is decoupled from k/k it will no longer benefit
+from all the updates and security fixes that its components under
+"staging" see. The imports in the kubeadm go.mod will need
+to be updated on a regular basis to mitigate CVEs and critical bugs.
+
+Tools such as https://docs.renovatebot.com/golang/ were
+proposed. Automatic bulk updates can increase the noise in the
+kubeadm repository but always make sure kubeadm depends on the
+latest dependencies. Manual updates can get sloppy and miss critical
+commits.
+
+The current proposal is to rely on manual updates and make sure
+kubeadm is always up to date before a final release, but also invest
+time into automating this process.
+
+Coordination with the security team and relevant SIGs might be
+required.
+
+#### The kubeadm module does not use SEMVER
+
+Users importing the kubeadm module will find that the k8s versioning
+is actually not SEMVER, because breaking changes happen on MINOR
+changes instead of MAJOR ones.
+
+A proposed solution here is to dual-tag commits with a version that
+is SEMVER friendly - e.g.:
+- `v1.17.0` -> `v0.17.0`
+
+Dual-tagging with `v17.0.0` is not an option, because Go modules
+will require the `v17` package.
+
+But the problem remains with `v0.17.0`, because once k8s becomes
+`v2.0.0`, the dual-tags (SEMVER) will overlap with the non-SEMVER tags.
+
+There are no good solutions for this problem and users might have to
+use SHAs to import the kubeadm module.
+
+See the related discussions at:
+https://github.com/kubernetes/kubernetes/issues/72638
+https://github.com/kubernetes/kubernetes/issues/84372
```

Comment:
If I read this correctly, https://github.com/kubernetes/kubernetes/issues/84372 indicates that the plan for the project broadly is to use the `v0.17.0` style for now as a stopgap, while we think of how to improve things?

This feels like it slightly conflicts with the earlier statement of:

```
The current plan is for kubeadm to continue to follow the k8s version scheme and release schedule. This will continue to make the skew of kubeadm and k8s easy to understand for the users and continue to give motivation to the kubeadm developers to get work done on time.
```

Is there consensus that users will understand v0.17.0 means v1.17.0?

Reply:
for now kubeadm will follow the exact versioning of k/k (and k8s).
the linked threads are examples where there are proposals of tagging differently repositories like k/client-go but the tagging schema there has issues.